docs/dev/multipath.md

```diff
@@ -0,0 +1,215 @@
+2. In SNO-SAN VM - "show virtual hardware details" -> Add Hardware -> Network -> select the isolated network you created in STEP-1 -> copy the MAC address
+3. In your beaker machine - 
+   - ```bash
+     sudo virsh net-edit SAN-network
+     ```
+   - In the dhcp section add the following:
+
+     ```xml
+     <host mac='<SAN mac from point (1)>' name='SAN' ip='192.168.100.125'/>
+     <host mac='<SNO-SAN mac from point (2)>' name='SNO-SAN' ip='192.168.100.126'/>
+     ```
+
+   - ```bash
+     sudo virsh net-destroy SAN-network
+      ```
+   - ```bash
+     sudo virsh net-start SAN-network
+     ```
+   - Check that the changes updated:
+     ```bash
+     sudo virsh net-dumpxml SAN-network
+     ```
+4. Click "finish" in points 1-2.
+5. In SNO-SAN VM -
+    ```bash
+    iscsiadm --mode node --targetname <iSCSI target you created in SAN> --portal 192.168.100.125:3260 -o new
+    iscsiadm --mode node --targetname<iSCSI target you created in SAN> --portal 192.168.100.125:3260 --login
+    ```
+6. Run `multipath -ll` to confirm that multipathing is active.
+
+## 2nd approach
+
+In this approach, your beaker machine will act as the target, while the initiator will be the test-infra VM.
+
+1. SSH into your beaker machine and navigate to the directory where the test-infra is cloned.
+2. Create a VM that includes two different networks:
+   ```bash
+   deploy_nodes_with_networking MASTERS_COUNT=1
+   ```
+3. On the target side (beaker machine):
+   - create disk on hypervisor:
+     ```bash
+     qemu-img create -o preallocation=full -f qcow2 /tmp/disk0 30G
+     ```
+   - Install tragetCli package:
+     ```bash
+     dnf install targetcli -y
+     ```
+     ```bash
+     targetcli clearconfig  confirm=True
+     targetcli backstores/fileio create name=disk0 size=30G file_or_dev=/tmp/disk0
+      ```
+   - Create a target:
+     ```bash
+     targetcli /iscsi create iqn.2023-01.com.example:target01
+     ```
+   - Create a LUN:
+     ```bash
+     targetcli /iscsi/iqn.2023-01.com.example:target01/tpg1/luns create /backstores/fileio/disk0
+     ```
+   - Create portal:
+     ```bash
+     targetcli /iscsi/iqn.2023-01.com.example:target01/tpg1/portals create <1st tearget IP>
+     targetcli /iscsi/iqn.2023-01.com.example:target01/tpg1/portals create <2nd tearget IP>
+     ```
+   - Allow access to initiator:
+     ```bash
+     targetcli /iscsi/iqn.2023-01.com.example:target01/tpg1/acls create iqn.1994-05.com.redhat:e1c30923a3a
+     targetcli / saveconfig
+     ```
+4. On the node side (test-infra VM):
+   - SSH into the VM you created in step 2.
+   - Discover (client):
+     ```bash
+     iscsiadm --mode discovery --type sendtargets --portal <1st target IP>
+     iscsiadm --mode discovery --type sendtargets --portal <2nd target IP>
+     ```
+   - Login to iSCSI:
+     ```bash
+     iscsiadm -m node -T iqn.2023-01.com.example:target01 -p <1st target IP>:3260 -l
```

Comment:
In the first option there was no discovery (why?) and the syntax was slightly different (e.g --portal and not -p). Please keep it consistent.
 
```
 iscsiadm --mode node --targetname <iSCSI target you created in SAN> --portal 192.168.100.125:3260 -o new
    iscsiadm --mode node --targetname<iSCSI target you created in SAN> --portal 192.168.100.125:3260 --login
```